src/ServiceControlInstaller.Engine/Instances/BaseService.cs

```diff
@@ -61,21 +61,27 @@ public bool TryStopService()
             try
             {
                 Service.WaitForStatus(ServiceControllerStatus.Stopped, TimeSpan.FromSeconds(60));
-                var t = new Task(() =>
-                {
-                    while (!HasUnderlyingProcessExited())
-                    {
-                        Thread.Sleep(100);
-                    }
-                });
-                t.Wait(5000);
             }
             catch (TimeoutException)
             {
                 return false;
             }
 
-            return true;
+            var processHasExited = false;
```

Comment:
Did you observe the previous code being wrong?

Reply:
Strike that, I see that it was wrong now. See my comment above

Reply:
I just saw this now. Are we OK here, or are you awaiting an answer to your question?

Reply:
Kind of, I was thinking that the original code was more clean? (assuming we fix the return)

Reply:
Ie https://github.com/Particular/ServiceControl/pull/1765#discussion_r335466963

Reply:
See my reply above, too. 😄